### PR TITLE
Move tools extension method into tools itself

### PR DESCRIPTION
**Pull request by josephdecock**
This simplifies the new interface considerably. And note that the new interface was introduced for v7, so this isn't a breaking change, apart from adding a new dependency to the ctor.

### CODE REVIEW

File: test/IdentityServer.UnitTests/Services/Default/DefaultBackChannelLogoutServiceTests.cs [hunk 1]
```diff
@@ -57,10 +57,10 @@ public async Task CreateTokenAsync_Should_Set_Issuer_Correctly()
         var issuerNameService = new TestIssuerNameService(expected);
 #pragma warning disable CS0618 // Type or member is obsolete
```
Comment (brockallen) [inline test/IdentityServer.UnitTests/Services/Default/DefaultBackChannelLogoutServiceTests.cs:58]: we can remove these pragmas then too

Comment (josephdecock) [reply test/IdentityServer.UnitTests/Services/Default/DefaultBackChannelLogoutServiceTests.cs:58]: Yeah, in the other PR I think.

**Comment by josephdecock** [comment]
@matt-goldman, this seems related to your pr: #1454.  Does refactoring in this way work for you?

**Comment by matt-goldman** [comment]
Does it make sense to reflect these changes in the `IIdentityServerTools` interface, given that they're not in the extensions interface now?

**Comment by josephdecock** [comment]
> Does it make sense to reflect these changes in the `IIdentityServerTools` interface, given that they're not in the extensions interface now?

Yes, absolutely, and Thanks!

**Pull request merged**